Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1049 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Securities;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using QuantConnect.Orders.Fees;
+using System.Threading;
+using QuantConnect.Orders;
+using QuantConnect.Brokerages.Zerodha.Messages;
+using Tick = QuantConnect.Data.Market.Tick;
+using System.Net.WebSockets;
+using System.Text;
+using Newtonsoft.Json.Linq;
+
+namespace QuantConnect.Brokerages.Zerodha
+{
+    /// <summary>
+    /// Zerodha Brokerage implementation
+    /// </summary>
+    [BrokerageFactory(typeof(ZerodhaBrokerageFactory))]
+    public partial class ZerodhaBrokerage : Brokerage, IDataQueueHandler, IDataQueueUniverseProvider, IHistoryProvider
```

Comment:
I think it would be much easier to **not** implement `IHistoryProvider` but only `IBrokerage.GetHistory` (called by the `BrokerageHistoryProvider` which already does all the work of synchronizing history for multiple symbols into slices)